saleor/graphql/discount/mutations.py

```diff
@@ -339,14 +342,18 @@ def clean_prices(cls, cleaned_input, voucher, errors):
             if missing_required_value or discount_value is None:
                 channels_without_value.append(cleaned_channel["channel_id"])
             # Validate value precision if it is fixed amount voucher
-            if discount_value and is_fixed_value_type:
+            if voucher.discount_value_type == DiscountValueType.FIXED:
                 try:
                     validate_price_precision(discount_value, channel.currency_code)
                 except ValidationError:
                     channels_with_invalid_value_precision.append(
                         cleaned_channel["channel_id"]
                     )
-            if discount_value:
+            elif voucher.discount_value_type == DiscountValueType.PERCENTAGE:
+                if discount_value > 100:
+                    channels_with_invalid_percentage_value.append(
+                        cleaned_channel["channel_id"]
+                    )
                 cleaned_channel["discount_value"] = discount_value
```

Comment:
Maybe I am missing something but it looks like you're passing `cleaned_channel["discount_value"` only when the type of discount is percent. Is that correct?

Reply:
That was unnecessary, i will remove it because `discount_value ` are already in cleaned input. Good catch, thanks.